### ISSUE TITLE
[Bug]: BrowserWindow.previewFile causes window to not update (appearing frozen)

### ISSUE THREAD

**Issue by jeremyspiegel** [electron/electron#37476]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

23.1.1, 24.0.0-alpha.6, 25.0.0-nightly-20230301

### What operating system are you using?

macOS

### Operating System Version

macOS 13.2.1

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Electron version

n/a

### Expected Behavior

When using `BrowserWindow.previewFile` and then choosing to open the file from QuickLook, the `BrowserWindow` content should update/respond normally.

### Actual Behavior

When using `BrowserWindow.previewFile` and then choosing to open the file from QuickLook, the `BrowserWindow` content stops updating visually. The window isn't completely frozen though, as javascript is still running and clicking on the page initiates event handlers. Timers seem to fire more slowly, however.

### Testcase Gist URL

https://gist.github.com/a15492e41ae5c4dacb1152c35bea5cc1

### Additional Information

The [testcase fiddle](https://gist.github.com/a15492e41ae5c4dacb1152c35bea5cc1) updates a button's text to a random string every 10 milliseconds, queries the button's text, and prints it to the console.

This video shows that after opening a .log file in Console.app via `BrowserWindow.previewFile`, the button text stops visually updating in the window, but it does update as shown in the devtools console. And clicking on the button causes a console message to be logged as well.

https://www.loom.com/share/4fbd0f43c61b47f3ab00acf64d2a196f

This test case uses Console.app, but we have users also reporting this issue for opening .pdf files in Acrobat Pro. It doesn't reproduce for me with .txt files opening TextEdit.

Reproducing the bug also seems to require that the window not be in the default position. It might also require moving the other application window (Console.app) after it's opened.

**Comment by jeremyspiegel** [comment]
In going through old Electron versions to see where this was introduced, I noticed that it seems to have disappeared in between 19.0.0-alpha.5 and 19.0.0-beta.1 (Chromium 102.0.4989.0 and 102.0.4999.0), so was not in stable Electron 19 releases but is in every other major version going back at least as far as Electron 11. There don't seem to be any relevant Electron changes between 19.0.0-alpha.5 and 19.0.0-beta.1 other than the 102.0.4999.0 update.

I also noticed that the problem goes away if you completely cover the window with another one, and then uncover it.

**Comment by jeremyspiegel** [comment]
I added logging for `document.hidden` and found that it becomes true when this happens.

**Comment by jeremyspiegel** [comment]
I believe I've figured out the problem: the QuickLook NSWindow's delegate is set to the same object as the delegate for the BrowserWindow's NSWindow [here](https://github.com/electron/electron/blob/f5385bcdec1951db34f8e55754265701530f5888/shell/browser/ui/cocoa/electron_ns_window.mm#L279). The delegate's [handler for windowDidChangeOcclusionState](https://github.com/electron/electron/blob/99f6ff430f57683ad0a240045c3cfbfbb8d67686/shell/browser/ui/cocoa/electron_ns_window_delegate.mm#L44) doesn't seem to expect handling events for the QuickLook window, and so when the QuickLook window becomes occluded, it assumes that the BrowserWindow has been occluded.

There's a similar issue for other `NSWindowDelegate` methods besides `windowDidChangeOcclusionState`. So resizing/moving the QuickLook window will cause the BrowserWindow's resize/will-resize/resized/move/will-move/moved events to fire.

The code has moved around a bit but I think this may have existed since `BrowserWindow.previewFile` was added in 2016 in https://github.com/electron/electron/pull/7592. FYI @lauracpierre @kevinsawicki @zcbenz @bengotow

I think the fix would be to remove the line where we set the QuickLook window's delegate, and will open a PR with that shortly.
